docs/operators/transform.md

```diff
@@ -0,0 +1,167 @@
+    
+#### Application Example
+------------------------
+
+Please refer [Example](https://github.com/DataTorrent/examples/tree/master/tutorials/transform) for transform sample application.
+
+### Partitioning
+Being stateless, this operator can be partitioned using any of the built-in partitioners present in the Malhar library by setting a few properties as follows:
+
+#### Stateless partitioning
+Stateless partitioning will ensure that TransformOperator will will be partitioned right at the starting of the application and will remain partitioned throughout the lifetime of the DAG.
+TransformOperator can be stateless partitioned by adding following lines to properties.xml:
+
+```xml
+  <property>
+    <name>dt.operator.{OperatorName}.attr.PARTITIONER</name>
+    <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}/value>
+  </property>
+```
+
+where {OperatorName} is the name of the TransformOperator operator and
+      {N} is the number of static partitions.
+Above lines will partition TransformOperator statically {N} times. 
+
+#### Dynamic Partitioning
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
+TransformOperator can be dynamically partitioned using the below two partitioners:
+
+##### Throughput based
+Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessThroughputBasedPartitioner<TransformOperator> partitioner = new StatelessThroughputBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
+    partitioner.setMaximumEvents(conf.getLong(MAX_THROUGHPUT, 30000));
+    partitioner.setMinimumEvents(conf.getLong(MIN_THROUGHPUT, 10000));
+    dag.setAttribute(transform, OperatorContext.STATS_LISTENERS, Arrays.asList(new StatsListener[]{partitioner}));
+    dag.setAttribute(transform, OperatorContext.PARTITIONER, partitioner);
+```
+
+Above code will dynamically partition TransformOperator when the throughput changes.
+If the overall throughput of TransformOperator goes beyond 30000 or less than 10000, the platform will repartition TransformOperator 
+to balance throughput of a single partition to be between 10000 and 30000.
+CooldownMillis of 10000 will be used as the threshold time for which the throughout change is observed.
+
+Source code for this dynamic application can be found [here](https://github.com/DataTorrent/examples/blob/master/tutorials/transform/src/main/java/com/example/transform/DynamicTransformApplication.java).
+
+##### Latency based
+Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to dynamically partitioning TransformOperator:
+```java
+    StatelessLatencyBasedPartitioner<TransformOperator> partitioner = new StatelessLatencyBasedPartitioner<>();
+    partitioner.setCooldownMillis(conf.getLong(COOL_DOWN_MILLIS, 10000));
```

Comment:
Please don't put conf.getLong... Its confusing... Just put number to setCoodownMillis and everything else.